web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js

```diff
@@ -25,8 +25,19 @@ odoo.define('web.web_ir_actions_act_window_message', function(require)
                     text: action.close_button_title || _t('Close'),
                     click: function() {
                         // refresh the view before closing the dialog
-                        self.inner_widget.active_view
-                            .controller.recursive_reload();
+                        var controller = self.inner_widget.active_view.controller;
+                        // recursive_reload function is available
+                        // in the form view
+                        if (controller.recursive_reload)
+                        {
+                            controller.recursive_reload();
+                        }
+                        // reload_content function is available
+                        // in the list view
+                        if (controller.reload_content)
+                        {
+                            controller.reload_content()
+                        }
```

Comment:
and in an `else` branch, you can `console.log` that a refresh was requested, that's nice for the next person using this in a different sort of view